test/decl/ext/extensions.swift

```diff
@@ -126,31 +126,102 @@ struct WrapperContext {
 }
 
 // Class-constrained extension where protocol does not impose class requirement
+// SR-11298
 
 protocol DoesNotImposeClassReq_1 {}
-	
+
 class JustAClass: DoesNotImposeClassReq_1 {
   var property: String = ""
 }
 
 extension DoesNotImposeClassReq_1 where Self: JustAClass {
-  var wrappingProperty: String {
+  var wrappingProperty1: String {
+    get { return property }
+    set { property = newValue } // Okay
+  }
+  
+  var wrappingProperty2: String {
+    get { return property }
+    nonmutating set { property = newValue } // Okay
+  }
+  
+  var wrappingProperty3: String {
     get { return property }
-    set { property = newValue }
+    mutating set { property = newValue } // Okay
+  }
+  
+  mutating func foo() {
+    property = "" // Okay
+    wrappingProperty1 = "" // Okay
+    wrappingProperty2 = "" // Okay
+    wrappingProperty3 = "" // Okay
+  }
+  
+  func bar() { // expected-note {{mark method 'mutating' to make 'self' mutable}}{{3-3=mutating }}
+    property = "" // Okay
+    wrappingProperty1 = "" // Okay
+    wrappingProperty2 = "" // Okay
+    wrappingProperty3 = "" // expected-error {{cannot assign to property: 'self' is immutable}}
+  }
+  
+  nonmutating func baz() { // expected-note {{mark method 'mutating' to make 'self' mutable}}
+    property = "" // Okay
+    wrappingProperty1 = "" // Okay
+    wrappingProperty2 = "" // Okay
+    wrappingProperty3 = "" // expected-error {{cannot assign to property: 'self' is immutable}}
   }
 }
-	
-let instanceOfJustAClass = JustAClass() // expected-note {{change 'let' to 'var' to make it mutable}}
-instanceOfJustAClass.wrappingProperty = "" // expected-error {{cannot assign to property: 'instanceOfJustAClass' is a 'let' constant}}
+
+let instanceOfJustAClass1 = JustAClass() // expected-note 2{{change 'let' to 'var' to make it mutable}}
+instanceOfJustAClass1.wrappingProperty1 = "" // Okay
+instanceOfJustAClass1.wrappingProperty2 = "" // Okay
+instanceOfJustAClass1.wrappingProperty3 = "" // expected-error {{cannot assign to property: 'instanceOfJustAClass1' is a 'let' constant}}
+instanceOfJustAClass1.foo() // expected-error {{cannot use mutating member on immutable value: 'instanceOfJustAClass1' is a 'let' constant}}
+instanceOfJustAClass1.bar() // Okay
+instanceOfJustAClass1.baz() // Okay
+
+var instanceOfJustAClass2 = JustAClass()
+instanceOfJustAClass2.foo() // Okay
 
 protocol DoesNotImposeClassReq_2 {
   var property: String { get set }
 }
 
 extension DoesNotImposeClassReq_2 where Self : AnyObject {
-  var wrappingProperty: String {
+  var wrappingProperty1: String {
     get { property }
-    set { property = newValue } // Okay
+    set { property = newValue } // expected-error {{cannot assign to property: 'self' is immutable}}
```

Comment:
As follow-up work, it would be nice to tweak the message and/or add a note to say *why* `self` is immutable.

Reply:
And if we could get a fix-it to insert `mutating` on the setter, we'd be making an easier migration path for the one place where behavior changed.

Reply:
Yeah, I've removed the old code that prevented the fix-it from being applied, it works now.